### DISCUSSION TITLE
How to set detection intervals ?

### DISCUSSION THREAD

**Discussion by utamori** [juliansteenbakker/mobile_scanner#39] [Q&A] [answered]
Thanks for the cool scanner. ❤️ 

I want an arbitrary interval to occur before the next execution of a function written to `onDetect`.
How can I do this?

I'm new to mobile development and Flutter, so sorry if I'm asking off the mark.

**Comment by juliansteenbakker** [comment, marked as answer]
You can work with intervals, but something that is more suitable is maybe to check on duplicates.

An interval can be something like 
```dart
  Timer? _timer; // Declared in your statefull class

// onDetect function of MobileScanner
    onDetect: (barcode, args) {
      if (_timer == null || !_timer!.isActive) {
        _timer = Timer(Duration(seconds: 2), () {});
        // do something
      }
  }
```

To check on duplicates you can do something like this:
```dart
String? barcode; // Declared in your statefull class

// onDetect function of MobileScanner
onDetect: (barcode, args) {
  if (this.barcode != barcode.rawValue) {
    // New barcode found !!
    setState(() {
      this.barcode = barcode.rawValue;
    });
  }
}
```

**Reply by utamori** [reply]
Thank you! 🎉 
Tried both and they worked perfectly.
Most of the time I'm fine with checking for duplicates, 
but there are times when I want to scan a series of ones with the same code, so I appreciate the first example too!
